src/Pact/Native/Internal.hs

```diff
@@ -114,6 +114,12 @@ defNative :: NativeDefName -> NativeFun e -> FunTypes (Term Name) -> [Example] -
 defNative n fun ftype examples docs =
   (n, TNative n (NativeDFun n (unsafeCoerce fun)) ftype examples docs False def)
 
+defGasNative ::  NativeDefName -> GasNativeFun e -> FunTypes (Term Name) -> [Example] -> Text -> NativeDef
```

Comment:
Why is this needed?

Reply:
I needed something like `defGasRNative` but with unreduced args since we take 2 lambdas in `fold-db`